### ISSUE TITLE
Issue with Inference – Mask Outputs Are All Black

### ISSUE THREAD

**Issue by SeanSun18** [SitongGong/VRS-HQ#8]
Thank you for sharing your work. I followed the provided inference steps in your repository [[repo name or link](https://github.com/SitongGong/VRS-HQ/issues/6#issuecomment-2841814787)], but the output mask images appear to be completely black. It seems the model is not generating any segmentation results. 

![Image](https://github.com/user-attachments/assets/9b06db6b-1e4f-48ac-9415-ad7d8842a744)

Here are my training/inference parameter settings:

![Image](https://github.com/user-attachments/assets/ac7bf710-5848-4715-94ff-e32de59ebf3a)

![Image](https://github.com/user-attachments/assets/771eaf91-f93f-4f49-9852-89b2cddc546d)

I’d appreciate any advice you could provide. Thank you again for your contribution!

**Comment by mohammedbennamoun** [comment]
I've spoken with a few people who attempted to deploy this model—it appears to be a challenging repository to set up. Making it easier to use could help increase citations.

**Comment by SitongGong** [comment]
@SeanSun18 Hi, have you encountered the following warning:

<img width="1266" alt="Image" src="https://github.com/user-attachments/assets/1f07ae50-652f-4904-b830-e59ad90c5f34" />

If so, it indicates that some weights in the memory encoder of SAM2 were not successfully loaded. You need to reload the model weights using load_state_dict.
In our code:

<img width="869" alt="Image" src="https://github.com/user-attachments/assets/67dcb1ed-0792-4b9c-a851-fc58e457d005" />

**Comment by SeanSun18** [comment]
@SitongGong Hi, thank you very much for your guidance!

I did comment out those two lines as you mentioned. The reason is that I encountered an error when trying to load the pytorch_model-00003-of-00003.bin file:
**PytorchStreamReader failed reading zip archive: failed finding central directory.**

Just now, I re-downloaded the model weights from the Baidu Netdisk link, but the same error still occurs.

Could you please help check whether the pytorch_model-00003-of-00003.bin file might be corrupted or not fully uploaded? Or is there anything I can do to resolve this issue?

I really appreciate your help and support!

**Comment by SitongGong** [comment]
Now the checkpoint of the Baidu desk has been rectified and the two lines should be adjusted as follows:
<img width="669" alt="Image" src="https://github.com/user-attachments/assets/1bb6264b-f882-4d0d-acd6-ef708fdf3166" /> Try running it again to see if the mask can now be generated correctly.

**Comment by SeanSun18** [comment]
@SitongGong Hi, thank you for your kind response.

Under your guidance, I encountered the following issue when loading the model:
![Image](https://github.com/user-attachments/assets/b30b4d23-57e7-4ed8-a545-c48b16e8e018)
It seems that this might be caused by a mismatch in the CLIP variant: the model was likely trained with **openai/clip-vit-large-patch14-224**, but during inference using **openai/clip-vit-large-patch14-336** instead. Could you please advise on the correct way to resolve this?

Additionally, I noticed that in **evaluation_multiseg.py**, the model is loaded using the **VISAForCausalLM** function, but in the header the import statement is commented out and replaced with **VrshqForCausalLM**. In my actual code, I used **VrshqForCausalLM consistently**. I'm wondering if this change might also be related to the issue above.
![Image](https://github.com/user-attachments/assets/6dc99dbf-0514-4be3-837e-4a9770a33732)
![Image](https://github.com/user-attachments/assets/18b8a591-d0c9-44a8-b5ed-f0dd1c3a28c3)

Thank you again for your continued support and guidance.

**Comment by SitongGong** [comment]
Hi, for your first question, the CLIP variant is openai/clip-vit-large-patch14-224 at both the training and inference stages.
Could you check whether the CLIP type you selected is correct?

<img width="380" alt="Image" src="https://github.com/user-attachments/assets/d99b95e2-5471-435a-b77d-9bcf433bd7f1" />

For your second question, the VISAForCausalLM should be rectified as VrshqForCausalLM.

**Comment by SeanSun18** [comment]
Thank you for your continued support.
